troposphere/haze/source/ptp_responder.cpp

```diff
@@ -700,4 +729,144 @@ namespace haze {
+                        R_TRY(db.Add(PtpDataTypeCode_U32));
+                        R_TRY(db.Add(PtpPropertyGetSetFlag_Get));
+                        R_TRY(db.Add(StorageId_SdmcFs));
+                    }
+                    break;
+                case PtpObjectPropertyCode_ObjectFormat:
+                    {
+                        R_TRY(db.Add(PtpDataTypeCode_U32));
+                        R_TRY(db.Add(PtpPropertyGetSetFlag_Get));
+                        R_TRY(db.Add(PtpObjectFormatCode_Undefined));
+                    }
+                    break;
+                case PtpObjectPropertyCode_ObjectFileName:
+                    {
+                        R_TRY(db.Add(PtpDataTypeCode_String));
+                        R_TRY(db.Add(PtpPropertyGetSetFlag_GetSet));
+                        R_TRY(db.AddString(""));
+                    }
+                    break;
+                HAZE_UNREACHABLE_DEFAULT_CASE();
+            }
+
+            /* Group code is a required part of the response, but doesn't seem to be used for anything. */
+            R_TRY(db.Add(PtpPropertyGroupCode_Default));
+
+            /* We don't use the form flag. */
+            R_TRY(db.Add(PtpPropertyFormFlag_None));
+
+            R_SUCCEED();
+        }));
+
+        /* We succeeded. */
+        R_RETURN(this->WriteResponse(PtpResponseCode_Ok));
+    }
+
+    Result PtpResponder::GetObjectPropValue(PtpDataParser &dp) {
+        u32 object_id;
+        PtpObjectPropertyCode property_code;
+
+        R_TRY(dp.Read(std::addressof(object_id)));
+        R_TRY(dp.Read(std::addressof(property_code)));
+        R_TRY(dp.Finalize());
+
+        /* Ensure we have a valid property code before continuing. */
+        R_UNLESS(IsSupportedObjectPropertyCode(property_code), haze::ResultUnknownPropertyCode());
+
+        /* Check if we know about the object. If we don't, it's an error. */
+        auto * const fileobj = m_object_database.GetObject(object_id);
+        R_UNLESS(fileobj != nullptr, haze::ResultObjectNotFound());
+
+        /* Get the object type. */
+        FsDirEntryType entry_type;
+        R_TRY(m_fs.GetEntryType(fileobj->GetName(), std::addressof(entry_type)));
+
+        /* Get the object size. */
+        s64 size = 0;
+        if (entry_type == FsDirEntryType_File) {
+            FsFile file;
+            R_TRY(m_fs.OpenFile(fileobj->GetName(), FsOpenMode_Read, std::addressof(file)));
+
+            /* Ensure we maintain a clean state on exit. */
+            ON_SCOPE_EXIT { m_fs.CloseFile(std::addressof(file)); };
+
+            R_TRY(m_fs.GetFileSize(std::addressof(file), std::addressof(size)));
+        }
+
+        /* Begin writing the requested object property. */
+        PtpDataBuilder db(g_bulk_write_buffer, std::addressof(m_usb_server));
+
+        R_TRY(db.WriteVariableLengthData(m_request_header, [&] {
+            switch (property_code) {
+                case PtpObjectPropertyCode_ObjectSize:
+                    R_TRY(db.Add<u64>(size));
```

Comment:
You do a GetFileSize() call regardless of whether the property code requested is size. Would it be better to save the FS calls and only retrieve the file size if it's needed?